internal/clients/kibana_oapi/data_views.go

```diff
@@ -87,3 +87,36 @@ func DeleteDataView(ctx context.Context, client *Client, spaceID string, viewID
 		return reportUnknownError(resp.StatusCode(), resp.Body)
 	}
 }
+
+// GetDefaultDataView reads the default data view from the API.
+func GetDefaultDataView(ctx context.Context, client *Client, spaceID string) (*string, diag.Diagnostics) {
+	resp, err := client.API.GetDefaultDataViewDefaultWithResponse(ctx, spaceID)
+	if err != nil {
+		return nil, diagutil.FrameworkDiagFromError(err)
+	}
+
+	switch resp.StatusCode() {
+	case http.StatusOK:
```

Comment:
I know that the schema doesn't define 404 response code but still does it make sense to explicitly handle it? Currently the call returns an error in this case but maybe we can treat 404 as a deleted resource, e.g. when space is deleted.